haystack/preview/components/fetchers/link_content.py

```diff
@@ -116,37 +117,71 @@ def from_dict(cls, data: Dict[str, Any]) -> "LinkContentFetcher":
-        Fetches content from a URL and converts it to a Document objects. If no content is extracted,
-        an empty Document object is returned (if raise_on_failure is False).
+        Fetches content from a list of URLs and returns a dictionary of extracted content streams.
+
+        :param urls: A list of URLs to fetch content from.
+
 
-        :param url: URL to fetch content from.
-        :param timeout: Timeout in seconds for the request.
-        :return: List of Document objects or an empty list if no content is extracted.
+        :return: A dictionary containing content streams categorized by content type.
+             The keys are content types (e.g., "text/html", "application/pdf"), and the values are lists of
+             ByteStream objects representing the extracted content.
         """
-        document_data: Dict[str, Any] = {"metadata": {"url": url, "timestamp": int(datetime.utcnow().timestamp())}}
+        streams: Dict[str, List[ByteStream]] = defaultdict(list)
+        if not urls:
+            return {"streams": streams}
+
+        # don't use multithreading if there's only one URL
+        if len(urls) == 1:
+            content_type, stream = self.fetch(urls[0])
+            if content_type and stream:
+                streams[content_type].append(stream)
+        else:
+            with ThreadPoolExecutor() as executor:
+                results = executor.map(self.fetch, urls)
+
+            for content_type, stream in results:
+                if content_type and stream:
+                    streams[content_type].append(stream)
+
+        return {"streams": streams}
+
+    def fetch(self, url: str) -> Tuple[str, ByteStream]:
+        """
+        Fetches content from a URL and returns it as a ByteStream.
+
+        :param url: The URL to fetch content from.
+        :return: A tuple containing the content type and the corresponding ByteStream.
+             The content type is a string indicating the type of content fetched (e.g., "text/html", "application/pdf").
+             The ByteStream object contains the fetched content as binary data.
+
+        :raises: If an error occurs during content retrieval and `raise_on_failure` is set to True, this method will
+        raise an exception. Otherwise, errors are logged, and an empty ByteStream is returned.
+
+        """
+        content_type: str = "text/html"
+        stream: ByteStream = ByteStream(data=b"")
         try:
             response = self._get_response(url)
             content_type = self._get_content_type(response)
-            document_data["mime_type"] = content_type
             handler: Callable = self.handlers[content_type]
-            document_data.update(handler(response))
-            return {"document": Document(**document_data)}
-
+            stream = handler(response)
         except Exception as e:
             if self.raise_on_failure:
                 raise e
             logger.debug("Couldn't retrieve content from %s", url)
```

Comment:
It would help users to understand the problem if we include `e` in the logging message too.